evm/vm_state.py

```diff
@@ -0,0 +1,299 @@
+from contextlib import contextmanager
+import copy
+import logging
+
+import rlp
+from trie import (
+    Trie,
+)
+
+from evm.db.tracked import (
+    AccessLogs,
+)
+
+
+class BaseVMState(object):
+    _chaindb = None
+    block_header = None
+    computation_class = None
+    access_logs = AccessLogs()
```

Comment:
This is probably problematic since the object will be shared across all instances of this class and the `access_logs` is mutable.  This should instead be initialized in the `__init__` method.